torch/distributed/checkpoint/_traverse.py

```diff
@@ -27,7 +28,7 @@
 __all__ = ["traverse_state_dict", "set_element", "get_element", "print_tensor"]
 
 
-def _keep_visiting_tensors(value: STATE_DICT_ITEM) -> bool:
+def _keep_visiting_tensors(value: STATE_DICT_ITEM) -> TypeIs[torch.Tensor]:
```

Comment:
Do we really want to do this? It's technically correct but doesn't really match the intent of the function (nothing about _keep_visiting_tensors seems to imply that your checking that the input is a tensor and indeed none of the uses are improved w/ this)